pkg/volume/flocker/plugin.go

```diff
@@ -199,9 +199,15 @@ func (b flockerBuilder) SetUpAt(dir string) error {
 		if err := b.updateDatasetPrimary(datasetID, primaryUUID); err != nil {
 			return err
 		}
+
+		newState, err := b.client.GetDatasetState(datasetID)
+		if err != nil {
+			return fmt.Errorf("The volume '%s' is not available in Flocker after migration.", dir)
+		}
+
+		b.flocker.path = newState.Path
 	}
 
```

Comment:
If <code>s.Primary == primaryUUID</code>, it means that we don't need to migrate the volume and the <code></code> would keep the original value.